src/widgets.zsh

```diff
@@ -47,6 +47,23 @@ _zsh_autosuggest_accept() {
 	_zsh_autosuggest_invoke_original_widget $@
 }
 
+# Accept the entire suggestion and execute it
+_zsh_autosuggest_execute() {
+	# Only accept if the cursor is at the end of the buffer
+	if [ $CURSOR -eq $#BUFFER ]; then
+		# Add the suggestion to the buffer
+		BUFFER="$BUFFER$POSTDISPLAY"
+
+		# Remove the suggestion
+		unset POSTDISPLAY
+
+		# Move the cursor to the end of the buffer
+		CURSOR=${#BUFFER}
+	fi
+	_zsh_autosuggest_invoke_original_widget $@
+	_zsh_autosuggest_invoke_original_widget accept-line
```

Comment:
I think we should be able to just go ahead and use `zle .accept-line` here instead